action.yaml

```diff
@@ -0,0 +1,196 @@
+name: Create installer
+description: Build the installer from a recipe directory.
+
+inputs:
+  conda-root:
+    description: Location to a pre-installed conda/mamba installation.
+    required: false
+  container-arch:
+    description: The architecture of the Docker image if different from the host.
+    required: false
+  container-image:
+    description: The Docker image used to build the installer.
+    required: false
+  environment-yaml-file:
+    description: |
+      Path to the environment.yaml file used to create the build environment.
+      Takes precedence over envrionemnt-yaml-string.
+      The file must not have the `name` property.
+    required: false
+  environment-yaml-string:
+    description: |
+      Multi-line string containing the contents of the environment.yaml file used to create the
+      build environment. Can be used to include logic and expressions from the calling workflow.
+      The file must not have the `name` property.
+    required: false
+  recipe-directory:
+    description: The directory containing the recipe file.
+    required: true
+  standalone-location:
+    description: The location of the standalone binary executable.
+    required: false
+
+  # secrets needed by constructor
+  azure-signtool-key-vault-client-id:
+    description: Client ID of the Azure key vault.
+    required: false
+  azure-signtool-key-vault-certificate:
+    description: Certificate of the Azure key vault.
+    required: false
+  azure-signtool-key-vault-secret:
+    description: Secret for the Azure key vault.
+    required: false
+  azure-signtool-key-vault-tenant-id:
+    description: Tenant ID of the Azure key vault.
+    required: false
+  azure-signtool-key-vault-url:
+    description: URL to for the Azure key vault.
+    required: false
+  constructor-pfx-certificate-password:
+    description: Windows signing password.
+    required: false
+
+outputs:
+  artifacts-directory:
+    description: Directory containing the artifacts of the constructor run.
+    value: ${{ steps.outputs.outputs.artifacts-directory }}
+
+runs:
+  using: composite
+  steps:
+    - name: Validate input
+      run: |
+        if [[ -z "${{ inputs.environment-yaml-file }}" ]] && [[ -z "${{ inputs.environment-yaml-string }}" ]]; then
+          echo "::error::\`environment-yaml-file\` or \`environment-yaml-string\` must be set."
+          exit 1
+        fi
+      shell: bash
+
+    - name: Set up workspace
+      id: workspace
+      run: |
+        UNAME="$(uname)"
+        CONSTRUCTOR_WORKSPACE="${{ runner.temp }}/_constructor_workdir"
+        if [[ "${UNAME}" == MINGW* ]]; then
+            CONSTRUCTOR_WORKSPACE=$(cygpath "${CONSTRUCTOR_WORKSPACE}")
+        fi
+        mkdir -p "${CONSTRUCTOR_WORKSPACE}"
+        chmod 777 "${CONSTRUCTOR_WORKSPACE}"
```

Comment:
Do we really need 777?

Reply:
Unfortunately, yes. For builds inside a container, the UID and GIDs are not the same for the runner and the container image, and you need `x` permissions to access that directory. If we want to mitigate the risks here somewhat, we can set those permissions only for those builds that use a container image.

Reply:
I changed it so that the default permissions are restored in the end to further mitigate risk.

Reply:
That step won't run if there's a failure. Added a comment below.